### PR TITLE
refactor: clean some code

### PR DESCRIPTION
**Pull request by liweitian**
## Description
use rimraf to replace deleteFilesFrombottomToUp

## Task Item

#minor

### CODE REVIEW

File: Composer/packages/server/src/models/storage/localDiskStorage.ts [hunk 1]
```diff
@@ -61,8 +63,12 @@ export class LocalDiskStorage implements IFileStorage {
...
 
-  async rmDir(path: string): Promise<void> {
-    await rmDir(path);
+  async rmDir(path: string, force?: boolean): Promise<void> {
+    if (force) {
+      await forceRmDir(path);
+    } else {
+      await rmDir(path);
+    }
   }
```
Comment (beyackle) [inline Composer/packages/server/src/models/storage/localDiskStorage.ts:72]: `rmdir` and `rm -rf` are different operations. `rm -rf` is not "force rmdir to happen", it's "remove a directory and everything inside it", and it ought to be its own method here. I'm fine with using `rimraf` as a way to implement it, but the boolean switch to choose between the two is not the way I'd go. `rmdir` should stay as pure `rmdir`. As a side benefit, this means you won't have to change the type of `rmDir` here, so those extra `force` arguments you added elsewhere can go away.

File: Composer/packages/server/src/models/bot/botProject.ts [hunk 2]
```diff
@@ -325,8 +325,7 @@ export class BotProject {
 
   public async deleteAllFiles(): Promise<boolean> {
     try {
-      await this.deleteFilesFromBottomToUp(this.dir);
-      await this.fileStorage.rmDir(this.dir);
+      await this.fileStorage.rmDir(this.dir, true);
```
Comment (beyackle) [inline Composer/packages/server/src/models/bot/botProject.ts:328]: This is where you'd call the new method (which can just be an invocation of `rimraf`; see the other comment).

File: Composer/packages/server/src/models/bot/botProject.ts [hunk 3]
```diff
@@ -363,31 +362,16 @@ export class BotProject {
...
-      const childStat = await this.fileStorage.stat(curPath);
-      if (childStat.isDir && curPath.startsWith(this.dir)) {
-        await this.deleteFilesFromBottomToUp(curPath);
-        await this.fileStorage.rmDir(curPath);
-      } else {
-        await this.fileStorage.removeFile(curPath);
-      }
-    }
-  }
-
```
Comment (beyackle) [inline Composer/packages/server/src/models/bot/botProject.ts:380]: 👍 to deleting this.

File: Composer/packages/server/src/models/storage/localDiskStorage.ts (hunk 1 above)
Comment (srinaath) [reply Composer/packages/server/src/models/storage/localDiskStorage.ts:72]: Do we need rmDir though @beyackle as it deletes just an empty directory. What if we change the function to removeDirectory and it accepts an options object like 
```
{
  recurse: true,
  force: true
}
```
and implement stuff based on the options object

**Comment by boydc2014** [comment]
Nice to see the cleanup. Can you take the chance to add more UT in the code you touched here if possible.

File: Composer/packages/server/src/models/bot/botProject.ts (hunk 3 above)
Comment (boydc2014) [reply Composer/packages/server/src/models/bot/botProject.ts:380]: +1, also can you also just delete AzureBLobStorage class which is not used and maintained anymore

File: Composer/packages/server/src/models/storage/localDiskStorage.ts (hunk 1 above)
Comment (liweitian) [reply Composer/packages/server/src/models/storage/localDiskStorage.ts:72]: will only use rimraf here to implement rmDir. I am adding some test cases to this pr.

Comment (boydc2014) [reply Composer/packages/server/src/models/storage/localDiskStorage.ts:72]: @liweitian i think what Ben is suggesting here is to explicitly add a interface for rimraf

**Comment by liweitian** [comment]
> Nice to see the cleanup. Can you take the chance to add more UT in the code you touched here if possible.

added UT to test delete all files function

File: Composer/packages/server/src/models/bot/botProject.ts (hunk 3 above)
Comment (liweitian) [reply Composer/packages/server/src/models/bot/botProject.ts:373]: deleted

File: Composer/packages/server/src/models/storage/localDiskStorage.ts (hunk 1 above)
Comment (liweitian) [reply Composer/packages/server/src/models/storage/localDiskStorage.ts]: add one more interface rmrfDir

**Pull request merged**